src/NzbDrone.Core/Download/Clients/Tribler/TriblerDownloadClient.cs

```diff
@@ -0,0 +1,327 @@
+                case DownloadDefaultsSeedingMode.Ratio:
+
+                    return torrent.Ratio.HasValue
+                        && torrent.Ratio >= config.Settings.DownloadDefaults.SeedingRatio;
+
+                case DownloadDefaultsSeedingMode.Time:
+                    var downloadStarted = DateTimeOffset.FromUnixTimeSeconds(torrent.TimeAdded.Value);
+                    var maxSeedingTime = TimeSpan.FromSeconds(config.Settings.DownloadDefaults.SeedingTime ?? 0);
+
+                    return torrent.TimeAdded.HasValue
+                        && downloadStarted.Add(maxSeedingTime) < DateTimeOffset.Now;
+
+                case DownloadDefaultsSeedingMode.Never:
+                    return true;
+
+                case DownloadDefaultsSeedingMode.Forever:
+                default:
+                    return false;
+            }
+        }
+
+        protected override string AddFromMagnetLink(RemoteEpisode remoteEpisode, string hash, string magnetLink)
+        {
+            var addDownloadRequestObject = new AddDownloadRequest
+            {
+                Destination = GetDownloadDirectory(),
+                Uri = magnetLink,
+                SafeSeeding = Settings.SafeSeeding,
+                AnonymityHops = Settings.AnonymityLevel
+            };
+
+            return _proxy.AddFromMagnetLink(Settings, addDownloadRequestObject);
+        }
+
+        protected override string AddFromTorrentFile(RemoteEpisode remoteEpisode, string hash, string filename, byte[] fileContent)
+        {
+            // tribler api currently do not support recieving a torrent file direcly.
+            throw new NotSupportedException("Tribler download client only support magnet links currently");
+        }
+
+        protected override void Test(List<ValidationFailure> failures)
+        {
+            failures.AddIfNotNull(TestConnection());
+            if (failures.HasErrors())
+            {
+                return;
+            }
+
+            // failures.AddIfNotNull(TestGetTorrents());
+        }
+
+        protected string GetDownloadDirectory()
+        {
+            if (Settings.TvDirectory.IsNotNullOrWhiteSpace())
+            {
+                return Settings.TvDirectory;
+            }
+
+            if (!Settings.TvCategory.IsNotNullOrWhiteSpace())
+            {
+                return null;
+            }
+
+            var config = _proxy.GetConfig(Settings);
+            var destDir = config.Settings.DownloadDefaults.SaveAS;
+
+            return $"{destDir.TrimEnd('/')}/{Settings.TvCategory}";
+        }
+
+        protected ValidationFailure TestConnection()
+        {
+            try
+            {
+                var downloads = GetItems();
+                return null;
+            }
+            catch (DownloadClientAuthenticationException ex)
+            {
+                _logger.Error(ex, ex.Message);
+
+                return new NzbDroneValidationFailure("ApiKey", "Authentication failure")
```

Comment:
validation failures should throw translated strings and detailed descriptions as they appear in the UI
(see other DL client test connection methods for examples)